Tweaks xeno chestbursting

benbot16 opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->
Someone suggested this to me at some point a while ago, but I never got around to coding it in (so sorry to whoever that was).
This PR makes it so that chestbursters won't gib their hosts. Instead, it'll just kill them and make all the organs in their chest spill out.
I also tweaked the logic on alien embryo growths to be a bit more consistent. Now, instead of checking prob(3) every time the egg processes, it just sets a timer from 1-1.5 minutes for each stage and progresses when that timer's up. If the embryo's removed, the stage timer resets. 
There's probably a better way to manage the stage times, but I'm not entirely sure if there's something else I should use. 

## Why It's Good For The Game

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding.
-->
Because full gibbing + brain deletion from a worm coming out of you never really made sense to me. That, and getting fully round removed kinda sucks.

The change for growth should also hopefully make embryo growth more consistent. 

## Testing Photographs and Procedure
<!-- Include any screenshots/videos/debugging steps of the modified code functioning successfully, ideally including edge cases. -->
It compiles, and there's no runtimes or incredibly broken things during embryo processing/chestbursting.

## Changelog
:cl:
tweak: Xeno chestbursters no longer gib their hosts.
tweak: Tweaks alien embryo growth logic
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

benbot16 committed:
Xeno chestbursts no longer gib + some embryo growth changes

benbot16 committed:
Something to handle a possible edge case

benbot16 committed:
Possible edge case fix

EvilDragonfiend commented:
People will not pick xeno if they believe they can still roundreturn. I think the braingib's purpose is that.

FriendlyContractor commented:
Bad idea. Bursted players are intentionally made unrevivable so they are encouraged to join as xenomorph and leaving them revivable will lead to various weird situations about which i will not go into detail here and most importantly can lead to degenerated situation where xenomorph hive can not grow because there are no players willing to take the role

Archanial commented on code/modules/mob/living/carbon/alien/special/alien_embryo.dm:
```diff
@@ -5,6 +5,7 @@
 	icon = 'icons/mob/alien.dmi'
 	icon_state = "larva0_dead"
 	var/stage = 0
+	var/next_stage_time = 0
```
Use cooldown defines

benbot16 committed:
Changes stage to use cooldown defines

Forgot a thing I needed existed in the cooldown defines, so I didn't use them originally. Changed them over to them since it exists.

williamcll commented:
Actually show accurate.

Archanial commented on code/modules/mob/living/carbon/alien/special/alien_embryo.dm:
```diff
@@ -50,24 +51,33 @@
+/obj/item/organ/body_egg/alien_embryo/on_death()
+	. = ..()
+	if(!owner && next_stage_time) // No growing outside the body
+		COOLDOWN_RESET(src, next_stage_time)
```
stop processing/kill/delete the organ instead

benbot16 replied:
This should do it.

benbot16 committed:
Changes embryo on_death handling
